### PR TITLE
Support internal and iTwin platform IMS authorities in integration tests

### PR DESCRIPTION
**Pull request by skirby1996**
This PR includes a number of bugfixes, and a new integration test pipeline in order to support defaulting our integration test CI builds to use the iTwin platform IMS authority (https://ims.bentley.com) and clients. The internal IMS authority and clients will still be tested with a new integration test pipeline which will trigger on updates to the master branch rather than as a check for PRs.

This PR also includes a few quality of life improvements to the ECSchema Full Stack RPC tests package.

### CODE REVIEW

File: core/backend-itwin-client/src/oidc/BackendAuthorizationClient.ts [hunk 1]
```diff
@@ -16,11 +16,13 @@ import { ClientMetadata, custom, Issuer, Client as OpenIdClient } from "openid-c
  */
 export interface BackendAuthorizationClientConfiguration {
   /** Client application's identifier as registered with the Bentley IMS OIDC/OAuth2 provider. */
-  clientId: string;
+  readonly clientId: string;
```
Comment (calebmshafer) [inline core/backend-itwin-client/src/oidc/BackendAuthorizationClient.ts:19]: @pbell97 FYI we'll want to pull this change over to the new client repo.

File: core/transformer/src/test/integration/IModelTransformerHub.test.ts [hunk 2]
```diff
@@ -381,7 +381,7 @@ describe("IModelTransformerHub (#integration)", () => {
 
     // create empty iModel meant to contain replayed master history
     const replayedIModelName = "Replayed";
-    const replayedIModelId = await IModelHost.hubAccess.createNewIModel({ iTwinId, iModelName: replayedIModelName, description: "blank", revision0: masterSeedFileName, noLocks: true });
+    const replayedIModelId = await IModelHost.hubAccess.createNewIModel({ iTwinId, iModelName: replayedIModelName, description: "blank", noLocks: true });
```
Comment (calebmshafer) [inline core/transformer/src/test/integration/IModelTransformerHub.test.ts:384]: Can you add the transformer integration tests to the main build if they aren't already?

File: clients/itwin/src/Client.ts [hunk 3]
```diff
@@ -113,6 +113,9 @@ export abstract class Client {
       } else {
         this._url = this.baseUrl;
       }
+
+      // Strip trailing '/'
+      this._url = this._url.replace(/\/$/, "");
```
Comment (calebmshafer) [inline clients/itwin/src/Client.ts:118]: Was this happening due to the `baseUrl` already having the `/` or just making it easier if someone set the env var with a trailing slash?

File: clients/reality-data/src/test/integration/RealityDataClient.test.ts [hunk 4]
```diff
@@ -106,6 +108,12 @@ describe("RealityServicesClient Normal (#integration)", () => {
   });
 
   it("should be able to retrieve the azure blob url (write access)", async () => {
+    // Skip this test if the issuing authority is not imsoidc.
+    // The iTwin Platform currently does not support the reality-data:write scope.
+    const imsUrl = await imsClient.getUrl(new ClientRequestContext(""));
+    if (-1 === imsUrl.indexOf("imsoidc"))
+      return;
```
Comment (calebmshafer) [inline clients/reality-data/src/test/integration/RealityDataClient.test.ts:115]: Can we do a `this.skip` here?  You may need to change the signature of the method to allow it from this,

```
async () => {
```

to this,

```
async function () => {
```

File: clients/itwin/src/Client.ts (hunk 3 above)
Comment (skirby1996) [reply clients/itwin/src/Client.ts:118]: Even though the ims authority env var was being set without a slash, when constructing the URL object and adding the prefix the value I was getting back included a trailing slash and this was just the simplest solution.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**